sdk/daml-lf/interpreter/src/test/scala/com/digitalasset/daml/lf/speedy/ExceptionTest.scala

```diff
@@ -45,12 +49,56 @@ class ExceptionTest(majorLanguageVersion: LanguageMajorVersion)
     SEApp(se, Array(SParty(p)))
   }
 
-  private val party = Party.assertFromString("Alice")
+  private val alice = Party.assertFromString("Alice")
+
+  private def runUpdateExpr(
+      compiledPackages: PureCompiledPackages,
+      expr: Expr,
+      getKey: PartialFunction[GlobalKeyWithMaintainers, Value.ContractId] = Map.empty,
+  ): Either[SError, SValue] = {
+    runUpdateExpr(
+      compiledPackages,
+      Map.empty,
+      compiledPackages.compiler.unsafeCompile(expr),
+      PartialFunction.empty,
+      getKey,
+    )
+  }
 
-  private def runUpdateExpr(pkgs1: PureCompiledPackages)(e: Expr): SResult[Question.Update] = {
-    def transactionSeed: crypto.Hash = crypto.Hash.hashPrivateKey("ExceptionTest.scala")
+  private def runUpdateApp(
+      compiledPackages: PureCompiledPackages,
+      packageResolution: Map[Ref.PackageName, Ref.PackageId],
+      expr: Expr,
+      args: Array[SValue],
+      getContract: PartialFunction[Value.ContractId, Value.VersionedContractInstance],
+      getKey: PartialFunction[GlobalKeyWithMaintainers, Value.ContractId],
+  ): Either[SError, SValue] = {
+    runUpdateExpr(
+      compiledPackages,
+      packageResolution,
+      SEApp(compiledPackages.compiler.unsafeCompile(expr), args),
```

Comment:
I am wondering it is safe, in case `args` is  empty here ?

Reply:
args has to match the arity of the compiled expression but I'm not too worried if it's unsafe because it's just a test helper